gadget/layout.go

```diff
@@ -515,7 +515,7 @@ func isLayoutCompatible(current, new *Volume) error {
 
 	// at the structure level we expect the volume to be identical
 	for i := range current.Structure {
-		if err := canUpdateStructure(current.Structure, i, new.Structure, i, new.Schema); err != nil {
+		if err := canUpdateStructure(current, i, new, i); err != nil {
 			return fmt.Errorf("incompatible structure #%d (%q) change: %v", new.Structure[i].YamlIndex, new.Structure[i].Name, err)
 		}
 	}
```

Comment:
do we need to test partial functionality also from the angle of gadget.IsCompatible? we also still have the question if all these functions can share more code, or we can have fewer functions?

Reply:
I've added some tests using gadget.IsCompatible as entry point. I definitely would like to another gadget refactoring round as a follow-up.